### PR TITLE
Add maximum circular subarray sum under Dynamic Programming

### PR DESCRIPTION
**Pull request by sumanbmondal**
### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [x] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: dynamic_programming/maximum_circular_subarray.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
+# import testmod for testing our function
+from doctest import testmod
```
Comment (CaedenPH) [inline dynamic_programming/maximum_circular_subarray.py:2]:
```suggestion
```

Why are you importing this twice?

**Review by CaedenPH — CHANGES_REQUESTED** [review]
Please could you link a source, such as a wikipedia article, that explains this algorithm

**Comment by sumanarlert** [comment]
Can someone please review the PR?

File: dynamic_programming/maximum_circular_subarray.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+    if min_so_far == arr_sum:
+        return max_so_far
+
+    # returning the maximum value
+    return max(max_so_far, arr_sum - min_so_far)
+
+
+if __name__ == "__main__":
+    import doctest
```
Comment (prateekiiest) [inline dynamic_programming/maximum_circular_subarray.py:43]: make the import outside main?

File: dynamic_programming/maximum_circular_subarray.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+    >>> max_circular_sum([8, -7, 9, -9, 10, -11, 12], 7)
+    23
+
+    >>> max_circular_sum([8, -7, 10, -9, 10, -11, 12], 7)
+    24
+    """
+
+    # Edge Case
+    if size == 1:
```
Comment (prateekiiest) [inline dynamic_programming/maximum_circular_subarray.py:16]: add test cases please

**Review by prateekiiest — CHANGES_REQUESTED** [review]

File: dynamic_programming/maximum_circular_subarray.py (hunk 3 above)
Comment (sumanarlert) [reply dynamic_programming/maximum_circular_subarray.py:22]: I had to the add test cases for the edge case right?

File: dynamic_programming/maximum_circular_subarray.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+import doctest
+
+
+# The function returns maximum circular contiguous sum in a[]
+def max_circular_sum(arr: list[int], size: int) -> int:
```
Comment (tianyizheng02) [inline dynamic_programming/maximum_circular_subarray.py:5]:
```suggestion
def max_circular_subarray_sum(arr: list[int], size: int) -> int:
```

File: dynamic_programming/maximum_circular_subarray.py
Comment (tianyizheng02) [inline dynamic_programming/maximum_circular_subarray.py:1]: `maximum_circular_subarray.py` -> `maximum_circular_subarray_sum.py` to be more accurate

**Review by tianyizheng02 — APPROVED** [review]

File: dynamic_programming/maximum_circular_subarray.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    23
+
+    >>> max_circular_sum([8, -7, 10, -9, 10, -11, 12], 7)
+    24
+
+    >>> max_circular_sum([-5], 1)
+    -5
+    """
+
+    # Edge Case
```
Comment (tianyizheng02) [inline dynamic_programming/maximum_circular_subarray.py:21]: What if the user enters 0 or a negative number for the size?

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]

**Review by tianyizheng02 — CHANGES_REQUESTED** [review]
Like I said during my last review, please make sure your code can correctly handle a 0 or negative size.

**Comment by tianyizheng02** [comment]
Just realized that there's no point in trying to fix this PR because **_IT'S PLAGIARIZED_**. If you take a look at the commits from before I tried to fix the code, you'll see that the code was taken directly from [this GeeksForGeeks page](https://www.geeksforgeeks.org/maximum-contiguous-circular-sum/). Honestly, it's on me that I didn't catch this sooner.
